nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/resources/docs/org.apache.nifi.atlas.reporting.AtlasNiFiFlowLineage/additionalDetails.html

```diff
@@ -0,0 +1,541 @@
+
+        <h4 id="nifi-atlas-types">NiFi Atlas Types</h4>
+
+        <p>This reporting task creates following NiFi specific types in Atlas Type system when it runs if these type definitions are not found.</p>
+
+        <p>Green boxes represent sub-types of DataSet and blue ones are sub-types of Process. Gray lines represent entity ownership.
+        Red lines represent lineage.</p>
+
+        <object data="nifi_types.svg" type="image/svg+xml" width="60%"></object>
+
+        <ul>
+            <li>nifi_flow
+                <p>Represents a NiFI data flow.</p>
+                <p>As shown in the above diagram, nifi_flow owns other nifi_component types.
+                    This owning relationship is defined by Atlas 'owned' constraint so that when a 'nifi_flow' entity is removed, all owned NiFi component entities are removed in cascading manner.</p>
+                <p>When this reporting task runs, it analyzes and traverse the entire flow structure, and create NiFi component entities in Atlas.
+                    At later runs, it compares the current flow structure with the one stored in Atlas to figure out if any changes has been made since the last time the flow was reported. The reporting task updates NiFi component entities in Atlas if needed.<p>
+                <p>NiFi components those are removed from a NiFi flow also get deleted from Atlas.
+                    However those entities can still be seen in Atlas search results or lineage graphs since Atlas uses 'Soft Delete' by default.
+                    See <a href="#delete-handler">Atlas Delete Handler</a> for further detail.</p>
+            </li>
+            Attributes:
+            <ul>
+                <li>qualifiedName: Root ProcessGroup ID@clusterName (e.g. 86420a14-2fab-3e1e-4331-fb6ab42f58e0@cl1)</li>
+                <li>name: Name of the Root ProcessGroup.</li>
+                <li>url: URL of the NiFi instance. This can be specified via reporting task 'NiFi URL for Atlas' property.</li>
+            </ul>
+        </ul>
+        <ul>
+            <li>nifi_flow_path <p>Part of a NiFi data flow containing one or more processing NiFi components such as Processors and RemoteGroupPorts. The reporting task divides a NiFi flow into multiple flow paths. See <a href="#path-separation">Path Separation Logic</a> for details.</p></li>
+            Attributes:
+            <ul>
+                <li>qualifiedName: The first NiFi component Id in a path@clusterName (e.g. 529e6722-9b49-3b66-9c94-00da9863ca2d@cl1)</li>
+                <li>name: NiFi component namess within a path are concatenated (e.g. GenerateFlowFile, PutFile, LogAttribute)</li>
+                <li>url: A deep link to the first NiFi component in corresponding NiFi UI</li>
+            </ul>
+        </ul>
+        <ul>
+            <!-- TODO: link to S2S details -->
```

Comment:
This was intended to be resolved?

Reply:
Forgot to delete the TODO. I was willing to write about the detail, but it's too internal I think. So just remove the TODO comment. I found other TODOs remaining in this document. All resolved.